/**:
  ros__parameters:

    # sim
    # NAMESPACE: /ego_racecar
    # OPPONENT_NAMESPACE: /opp_racecar
    # GLOBAL_PATH_TOPIC: /ego_racecar/waypoints
    # DRIVE_TOPIC: /drive

    # physical
    NAMESPACE: /ego_racecar
    OPPONENT_NAMESPACE: /opp_racecar
    GLOBAL_PATH_TOPIC: /path
    DRIVE_TOPIC: /drive
    ODOM_TOPIC: /pf/pose/odom
    SPEED_TOPIC: /target_speed
    SELECTED_PATH_TOPIC: /ego_racecar/selected_path

    COLOR: green

    STEERING_INCREMENT: 0.6109
    
    # ACCELERATION_INCREMENT: 0.25
    ACCELERATION_INCREMENT: 0.5

    ACTION_LST: [-1.0, -1.0, 
                 -1.0, 0.0,
                 -1.0, 1.0, 
                  0.0, -1.0,
                  0.0, 0.0,
                  0.0, 1.0,
                  1.0, -1.0, 
                  1.0, 0.0,
                  1.0, 1.0, 
                  1.0, -0.5,
                  1.0, 0.5]
    
    # e7_floor5
    # MAX_SPEED: 1.0  # m/s
    # ACTION_INTERVAL: 0.25 # seconds
    # RECEDING_HORIZON: 1.5 # seconds
    # DT: 0.25

    # hallway - scalar parameters
    IS_VECTOR_COST: False
    PROGRESS_RESET_RATIO: 0.75
    # MAX_SPEED: 1.0  # m/s
    MAX_SPEED: 0.6  # m/s
    ACTION_INTERVAL: 0.1 # seconds
    RECEDING_HORIZON: 1.5 # seconds
    DT: 0.25

    # hallway - vector parameters
    # IS_VECTOR_COST: True
    # PROGRESS_RESET_RATIO: 0.8
    # MAX_SPEED: 0.6  # m/s
    # ACTION_INTERVAL: 0.1 # seconds
    # RECEDING_HORIZON: 5.0 # seconds
    # DT: 0.5

    # PROGRESS_RESET_RATIO: 0.75
    # MAX_SPEED: 0.6  # m/s
    # ACTION_INTERVAL: 0.1 # seconds
    # RECEDING_HORIZON: 3.0 # seconds
    # DT: 0.5

    # size for calculations, radial (width) and frontal (length) axes
    LF: 1
    LR: 1

    # how close bike center points are in pixels to count as collision
    COLLISION_RADIUS: 2.0
    PROXIMITY_SPREAD: 1.0
    BOUNDS_SPREAD: 3.0

    PROGRESS_WEIGHT: 1.0
    BOUNDS_WEIGHT: 4.0
    PROXIMITY_WEIGHT: 1.0

    OPP_COLLISION_RADIUS: 2.0
    OPP_PROXIMITY_SPREAD: 1.0
    OPP_BOUNDS_SPREAD: 3.0

    OPP_PROGRESS_WEIGHT: 1.0
    OPP_BOUNDS_WEIGHT: 4.0
    OPP_PROXIMITY_WEIGHT: 1.0